server/ot_rpc.go

```diff
@@ -304,14 +317,20 @@ func (h *lphttp) TranscodeResults(w http.ResponseWriter, r *http.Request) {
 				break
 			}
 
-			encodedPixels, err := strconv.ParseInt(p.Header.Get("Pixels"), 10, 64)
-			if err != nil {
-				glog.Error("Error getting pixels in header:", err)
-				res.Err = err
-				break
+			if len(p.Header.Values("Pixels")) > 0 {
+				encodedPixels, err := strconv.ParseInt(p.Header.Get("Pixels"), 10, 64)
+				if err != nil {
+					glog.Error("Error getting pixels in header:", err)
+					res.Err = err
+					break
+				}
+				segments = append(segments, &core.TranscodedSegmentData{Data: body, Pixels: encodedPixels})
+			} else if p.Header.Get("Content-Type") == "application/octet-stream" {
+				// Perceptual hash data for last segment
+				seg := *segments[len(segments)-1]
```

Comment:
Why not just
`segments[len(segments)-1].PHash = body` ? Array contains pointer, so it should work?
Also, if somehow transcoder will send wrong data, then array `segments` could be empty and thus it will crash.
Shouldn't we check array's length?

Reply:
Fixed